src/setup/validateTxnFields.ts

```diff
@@ -247,6 +247,18 @@ export const validateTxnFields =
           reason = ''
         }
 
+        // Chain ID validation
+        let chainId = BigInt(-1)
+        if (transaction && transaction.common.chainId) {
+          chainId = transaction.common.chainId()
+        }
+        if (chainId === BigInt(-1) || chainId !== BigInt(ShardeumFlags.ChainID)) {
+          nestedCountersInstance.countEvent('shardeum', 'validate - invalid chain ID')
+          success = false
```

Comment:
Do we not need to return this?

Reply:
According to the current flow, not required. In the end, we return success, reason, and timestamp.